src/components/LeftSidebar/LeftSidebar.vue

```diff
@@ -668,16 +680,41 @@ export default {
 				 * Emits a global event that is used in App.vue to update the page title once the
 				 * ( if the current route is a conversation and once the conversations are received)
 				 */
-				EventBus.$emit('conversations-received', {
-					singleConversation: false,
-				})
+				EventBus.$emit('conversations-received', { singleConversation: false })
 				this.isFetchingConversations = false
 			} catch (error) {
 				console.debug('Error while fetching conversations: ', error)
 				this.isFetchingConversations = false
 			}
 		},
 
+		async restoreConversations() {
+			try {
+				await this.$store.dispatch('restoreConversations')
+				this.initialisedConversations = true
+				EventBus.$emit('conversations-received', { singleConversation: false })
+			} catch (error) {
+				console.debug('Error while restoring conversations: ', error)
+			}
+		},
+
+		/**
+		 * Assign the first window, which requested that, a 'fetching leader'.
+		 * It will fetch conversations with defined interval and cache to BrowserStorage
+		 * Other will update list with defined interval from the BrowserStorage
+		 * Once 'leader' is closed, next requested tab will be assigned, and so on
+		 */
+		async onFetchingConversationsLeader() {
+			return new Promise((resolve) => {
+				window.navigator.locks.request('talk:conversations-fetching-leader', () => {
+					resolve()
+
+					// return an infinity promise, resource is blocked until 'leader' tab is closed
+					return new Promise(() => {})
+				})
+			})
+		},
```

Comment:
**issue (blocking):** We should also check for lock availability here and consider any tab to be a leader otherwise. It is not available without HTTPS.